### PR TITLE
Cleanup Batching Implementation

### LINKED ISSUE #5756: Cleanup Batching Implementation
The batching implementation could be cleaned up in some areas.

- [ ] Remove the tight `while` loop
  - [ ] Remove the Checkstyle suppression once it is done
- [ ] Move where the response is set to retrieved
- [x] Remove unnecessary portions in the `pom.xml`
- [x] Add format templates for commonly used templates
- [x] Fix ArrayIndexOutOfBoundsException when splitting strings
  * Examples
    * https://github.com/Azure/azure-sdk-for-java/pull/5734/files#r333257305
    * https://github.com/Azure/azure-sdk-for-java/pull/5734/files#r333260091
- [x] Update comments that say URI
- [x] Remove duplicate `HttpPipeline.getPolicy()` call
- [x] Use default `AtomicInteger` constructor
- [x] Use `Mono.fromRunnable` to make request construction asynchronous
- [x] Cleanup links using `en-us` in the URL

### PR DESCRIPTION
**Pull request by alzimmermsft**
Fixes #5756 

Cleans up some implementation details of Blob Batching. Changes the exception handling to throw a `BlobStorageException` on request failure and `StorageBlobBatchException` which is an aggregate of the batch operation exceptions.

### CODE REVIEW

**Comment by JonathanGiles** [comment]
Should there be naming consistency in the exceptions? You have BlobStorageException and StorageBlobBatchException

**Comment by alzimmermsft** [comment]
@JonathanGiles thoughts on BlobBatchStorageException

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/StorageBlobBatchException.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+        this.causes = causes;
+    }
+
+    /**
+     * Gets all the exceptions thrown in a single batch request.
+     *
+     * @return All the exceptions thrown in a single batch request.
+     */
+    public Iterable<BlobStorageException> getCauses() {
```
Comment (JonathanGiles) [inline sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/StorageBlobBatchException.java]: I'm not sure I would know what 'causes' means in this context. I would expect it to be called something more intuitive - getBatchExceptions() or getExceptions() or similar?

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/StorageBlobBatchException.java]: Changed to `getBatchExceptions`

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java [hunk 2]
```diff
@@ -391,34 +397,34 @@ String getContentType() {
...
+            appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TYPE);
+            appendWithNewline(batchRequestBuilder, BATCH_OPERATION_CONTENT_TRANSFER_ENCODING);
+            appendWithNewline(batchRequestBuilder, String.format(BATCH_OPERATION_CONTENT_ID_TEMPLATE, contentId));
+            batchRequestBuilder.append(BlobBatchHelper.HTTP_NEWLINE);
+
+            String method = request.getHttpMethod().toString();
+            String urlPath = request.getUrl().getPath();
+            String urlQuery = request.getUrl().getQuery();
+            if (!ImplUtils.isNullOrEmpty(urlQuery)) {
+                urlPath = urlPath + "?" + urlQuery;
```
Comment (conniey) [inline sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java]: nit: `urlPath += "?" + urlQuery;` ?

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java [hunk 3]
```diff
@@ -47,16 +60,33 @@
...
 
         return FluxUtil.collectBytesInByteBufferStream(batchResponse.getValue())
-            .flatMap(byteArrayBody -> {
+            .flatMap(byteArrayBody -> Mono.fromRunnable(() -> {
                 String body = new String(byteArrayBody, StandardCharsets.UTF_8);
+                List<BlobStorageException> exceptions = new ArrayList<>();
+
+                String[] subResponses = body.split("--" + boundary);
+                if (subResponses.length == 3 && batch.getOperationCount() != 1) {
+                    String[] exceptionSections = subResponses[1].split(HTTP_NEWLINE + HTTP_NEWLINE);
```
Comment (conniey) [inline sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java]: Some of these I'm confused about, and may require some docs like.. why do we split on two new lines?

Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java]: I'll comment this, HTTP responses split their metadata and body with `\r\n\r\n`.

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java]: I will add an issue to change this next week as it won't touch API

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java]: I will add an issue to change this next week as it won't touch API

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java]: https://github.com/Azure/azure-sdk-for-java/issues/5942

File: sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java (hunk 3 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatchHelper.java]: https://github.com/Azure/azure-sdk-for-java/issues/5942

**Pull request merged**